### DISCUSSION TITLE
build ANE lib

### DISCUSSION THREAD

**Discussion by pierrehub2b** [airsdk/Adobe-Runtime-Support#2980] [General]
Hello, I have made a very simple C code file 'KeyHook.c' and compile to a dll with this command line : 

`cl /D_USRDLL /D_WINDLL KeyHook.c /link win64/FlashRuntimeExtensions.lib /DLL /OUT:bin/KeyHook64.dll`

I have created an actionscript library project with main class 'KeyHook.as' and build the swc using this ant task : 

```
		<compc compiler.as3="true" 
			compiler.debug="false" 
			compiler.optimize="true" 
			compiler.accessible="true" 
			output="bin/KeyHook.swc" 
			incremental="false" 
			fork="true" 
			maxmemory="512m">
					<external-library-path file="${FLEX_HOME}/frameworks/libs/air/airglobal.swc" append="true"/>
					<external-library-path file="${FLEX_HOME}/frameworks/libs/air/aircore.swc" append="true"/>
					<external-library-path file="${FLEX_HOME}/frameworks/libs/air/airspark.swc" append="true"/>
					<external-library-path file="${FLEX_HOME}/frameworks/libs/air/airframework.swc" append="true"/>
					<external-library-path file="${FLEX_HOME}/frameworks/libs/air/gamepad.swc" append="true"/>
					<external-library-path file="${FLEX_HOME}/frameworks/libs/air/servicemonitor.swc" append="true"/>
					<include-sources dir="src" includes="*" />
		</compc>
```

I have extracted the 'library.swf' from the 'KeyHook.swc' generated library.

Finally I launch the following command line : 
`adt.bat -package -target ane KeyHook64.ane extension64.xml -swc KeyHook.swc -platform Windows-x86-64 library.swf KeyHook64.dll`

But this command fail with the log : Missing 'library.swf' for platform: Windows-x86-64

Can you help me ? (I only need the x64 version of the ANE)

**Comment by ajwfrost** [comment]
I can't see anything wrong with that .. assuming your library.swf and DLL are in the same folder as your XML and SWC files, it should work..! Although, we do usually put the platform files into a subfolder so our build commands are more like:
```
adt.bat -package -target ane KeyHook64.ane extension64.xml -swc KeyHook.swc -platform Windows-x86-64 -C win64 .
```
Just out of interest, presumably your extension XML doesn't mention any default platform or anything other than Windows-x86-64? am just wondering if it's expecting something else, and the error message is poor. Worth also checking that the library.swf is actually valid (i.e. not ended up accidentally zero-bytes, or still open/locked by a zip extractor, etc)?

thanks

**Reply by pierrehub2b** [reply]
Hello Andrew,

Thank you very much for your fast reply and suggestion.
I have try with your adt command line and now it works !

Thanks again for this christmas gift :-))
